pkg/storage/tsdb/block/index.go

```diff
@@ -32,8 +34,18 @@ import (
 )
 
 // VerifyIndex does a full run over a block index and verifies that it fulfills the order invariants.
-func VerifyIndex(logger log.Logger, fn string, minTime, maxTime int64) error {
-	stats, err := GatherIndexHealthStats(logger, fn, minTime, maxTime)
+func VerifyIndex(logger log.Logger, blockDir string, minTime, maxTime int64) error {
+	stats, err := GatherBlockHealthStats(logger, blockDir, minTime, maxTime, false)
+	if err != nil {
+		return err
+	}
+
+	return stats.AnyErr()
+}
+
```

Comment:
Let's remove `VerifyIndex` and only keep `VerifyBlock`, and add `checkChunks` boolean flag to it, similar to what the PR does for `GatherBlockHealthStats`.